Sources/HAP/Endpoints/pairSetup().swift

```diff
@@ -38,36 +38,44 @@ func pairSetup(device: Device) -> Application {
         var body = Data()
         guard
             (try? request.readAllData(into: &body)) != nil,
-            let data: PairTagTLV8 = try? decode(body),
+            let data: [PairTagTLV8] = try? decode(body),
             let sequence = data[.state]?.first.flatMap({ PairSetupStep(rawValue: $0) })
         else {
             return .badRequest
         }
-        let response: PairTagTLV8?
+        let response: [PairTagTLV8]?
         do {
             switch sequence {
+            // M1: iOS Device -> Accessory -- `SRP Start Request'
             case .startRequest:
                 let session = createSession()
                 response = try controller.startRequest(data, session)
                 connection.context[SESSION_KEY] = session
+            // M3: iOS Device -> Accessory -- `SRP Verify Request'
             case .verifyRequest:
                 let session = try getSession(connection)
                 response = controller.verifyRequest(data, session)
+            // M5: iOS Device -> Accessory -- `Exchange Request'
             case .keyExchangeRequest:
                 let session = try getSession(connection)
                 response = try controller.keyExchangeRequest(data, session)
+            // Unknown state - return error and abort
             default:
-                response = nil
+                response = try controller.unknownRequest(data)
             }
         } catch {
             logger.warning(error)
             connection.context[SESSION_KEY] = nil
             response = nil
         }
         if let response = response {
+            if let errorData = response[.error] {
```

Comment:
I rather stick with throwing errors from the controller, which can be caught in the endpoint and then converted to the various error responses as necessary.

Reply:
OK I've addressed that in the latest commit.